### PR TITLE
[SDF-46] Group (prefab) export/import backend

### PR DESCRIPTION
**Pull request by migoox**
Overview:
1. Provided a JSON Schema that documents the prefab format. When merged, the JSON can be referenced with url in order to validate the prefab format. All definitions are stored in a separate file (`definitions.schema.json`) as they will be useful for scene format in the future.
2. Added JSON validation library (`valijson`) in order to validate the JSON in tests. I didn't add it to deserialization functions, as the code I've written already validates the JSON. The cmake `inject_defs_into_schema` function merges the `definitions.schema.json` and `prefab.schema.json` into one json that is pasted into `json_schemas.hpp`.
3. Improved the enum mapper.

### CODE REVIEW

**Review by adamieck — APPROVED** [review]
fire 🔥

**Review by kubazuch — APPROVED** [review]
Lgtm

File: libresin/libresin/utils/enum_mapper.hpp
[the remark this answers is not in the document]
Comment (migoox) [reply libresin/libresin/utils/enum_mapper.hpp:1]: fav lang 🗿

File: libresin/libresin/core/sdf_tree/sdf_tree.cpp
[the remark this answers is not in the document]
Comment (migoox) [reply libresin/libresin/core/sdf_tree/sdf_tree.cpp:1]: I also have mixed feelings, but after copying third function body, the "they do reduce the boilerplate" argument won

File: libresin/tests/random_helper.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include <glm/fwd.hpp>
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <libresin/core/material.hpp>
+#include <libresin/core/transform.hpp>
+#include <random>
+
+inline std::mt19937& get_random_gen() {
+  static std::mt19937 generator(42);
```
[the remark this answers is not in the document]
Comment (migoox) [reply libresin/tests/random_helper.hpp:9]: tbh I didn't see any arguments against it, if you know any, it should be easy to change,  it simplifies the debugging as you said, maybe when more sophisticated randomizer arrive we should consider testing several seeds

File: libresin/tests/random_helper.hpp
[the remark this answers is not in the document]
Comment (migoox) [reply libresin/tests/random_helper.hpp:1]: thus the name 😊

File: libresin/tests/random_helper.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kubazuch) [reply libresin/tests/random_helper.hpp:9]: I mean we're not creating a game so we don't need sophisticated randomness 😸

**Pull request merged**